tile/platform/local_machine/mem_cache.cc

```diff
@@ -8,6 +8,25 @@ namespace vertexai {
 namespace tile {
 namespace local_machine {
 
+MemCache::MemCache(const std::shared_ptr<DevInfo>& dev_info, hal::Memory* source) :
+  total_{0}, source_{source} {
+  mem_limit_ = dev_info->settings.max_global_mem();
+}
+
+void MemCache::Cleanup(size_t new_size) {
+  while (total_ + new_size > mem_limit_) {
+    // Out of device memory, clean up cache
+    while (mem_.size() > 0 && mem_.begin()->second.empty()) {
```

Comment:
Fun trick: if you consistently order comparisons s.t. you're always using '<', it makes it just a little easier to think of things as existing on a number line, and makes it just a little harder to get your comparisons wrong.  Not as big a deal for something this simple (you should just remove the comparison), but I think it's a good idea in general.